Packages/com.onchainstudios.uitoolkitextensions/Runtime/Scripts/UIDocumentEventBusBridge/UIDocumentEventBusBridge.cs

```diff
@@ -41,5 +41,17 @@ protected virtual void OnDisable()
                 VisualElementCallbackManager.UnregisterCallbacks(UIDocument.rootVisualElement);
             }
         }
+
+        /// <summary>
+        /// Registers callbacks, clearing previously registred callbacks
+        /// </summary>
+        public void RegisterCallbacks()

```

Comment:
Cleanup recommendation:
1. Make a property called IsRootVisualElementValid and replace all cases of UIDocument != null && UIDocument.rootVisualElement != null with that.

2. Make 3 methods
M1: protected RegisterCallbacks
M2: protected UnregisterCallbacks
M3: public ReregisterCallbacks

Call Register and Unregister from OnEnable and OnDisable like so
If (IsRootVisualElementValid)
{
   RegisterCallbacks()
}

Reregister should call register and unregister respectively. This will encapsulate the logic in a less redundant way.